test(e2e): allow webkit-only 'access control checks' pageerror on changelog popup teardown

## What

`test_password_change_hx_refresh_recovers_csrf[webkit]` intermittently fails in **teardown** (the autouse `console_guard` fixture in `tests/e2e_browser/conftest.py`) with:

```
AssertionError: Unexpected console / page errors:
['pageerror: /localhost:8877/settings/changelog/popup due to access control checks.']
```

Chromium and firefox do not surface this error. The test itself passes; only the post-test console assertion fires.

## Why

The successful password change returns `HX-Refresh: true`, which forces a full page reload. WebKit reports `access control checks` page-level errors when an in-flight fetch (the changelog popup HTMX request, fired by the previous page) hits a session-cookie rotation in the middle of a navigation. The same family of webkit-only quirks is already documented in `conftest.py` for bare `htmx:*` event names; this is a sibling case the existing allowlist does not match.

## Proposed change

Extend `_ALLOWED_ERROR_PATTERNS` in `tests/e2e_browser/conftest.py` with a webkit-only pattern matching `pageerror: ... due to access control checks.` and add a comment explaining the trigger (HX-Refresh while a sibling fetch is in flight, only on webkit).

## Acceptance criteria

- The webkit job stops failing on this teardown when the only flake source is the access-control pageerror.
- Chromium / firefox behavior unchanged.
- Other unexpected page errors still fail the test (no broad regex that swallows real bugs).

## Out of scope

- Suppressing real authentication bugs.
- Reworking the changelog popup endpoint or its request lifecycle.

--- a/tests/e2e_browser/conftest.py
+++ b/tests/e2e_browser/conftest.py
@@ -39,6 +39,20 @@ _ALLOWED_ERROR_PATTERNS = [
     # mid-flight; allow the bare event-name console errors so the
     # autouse console_guard does not flag the teardown on webkit.
     re.compile(r"^htmx:[A-Za-z]+(?:Error|Request|Swap|Send|Response)?$"),
+    # Sibling of the bare-event-name pattern above, but at the
+    # ``pageerror`` (window-level) layer instead of console.error.
+    # Webkit reports any in-flight fetch canceled by a pending
+    # navigation as ``<URL> due to access control checks.`` even on
+    # same-origin requests.  The recurring trigger for this suite is
+    # the changelog popup's ``hx-trigger="load"`` GET still being in
+    # flight when ``HX-Refresh`` after a password change calls
+    # ``location.reload()``.  No client-side mitigation exists (the
+    # browser fires the error on its own abort path), and chromium /
+    # firefox swallow the same abort silently.  Documented upstream
+    # in TanStack/router#719 and supabase/supabase#20982.  Scope the
+    # match to the changelog popup path so a real auth failure on
+    # any other endpoint still fails the teardown.
+    re.compile(r"^pageerror:\s.*/settings/changelog/popup\b[^\n]*access control checks\.?$"),
 ]
 
 

